src/base.py

```diff
@@ -152,11 +147,217 @@ def __init__(self, **kwargs):
             raise ValueError(f"dim = {self.dim} not supported")
         
         # Compute the bounding box indices for boundary conditions
-        self.boundingBoxIndices = self.bounding_box_indices()
+        self.boundingBoxIndices= self.bounding_box_indices()
         # Create boundary data for the simulation
         self._create_boundary_data()
         self.force = self.get_force()
 
+    @property
+    def lattice(self):
+        return self._lattice
+
+    @lattice.setter
+    def lattice(self, value):
+        if value is None:
+            if isinstance(self, src.models.BGKSim):
```

Comment:
I think we should raise error inside KBC model class if D3Q19 is explicitly used. 
Also I don't think it would be reasonable to completely remove lattice definition in all examples. This addition here is useful in-case someone "forgets" to identify the lattice but we should keep its definition inside examples EXPLICIT. This is mainly because the current defaults imply that BGK cannot be used with D3Q27 for example.